skmediate/conditional_independence.py

```diff
@@ -81,28 +144,29 @@ def fit(self, X, Z, Y):
         # Step 1: Residualize with regression
 
         # TODO: Check regression type for supporting single or multi-output regression
-        regfit_xy = self.regression_estimator_xy.fit(X, Y)
         regfit_xz = self.regression_estimator_xz.fit(X, Z)
+        regfit_xy = self.regression_estimator_xy.fit(X, Y)
 
         # Compute residualized Zs and Ys.
         self.residualized_Z_ = Z - regfit_xz.predict(X)
         self.residualized_Y_ = Y - regfit_xy.predict(X)
 
         # Step 2: Covariance estimation
-        W = np.concatenate((self.residualized_Z_, self.residualized_Y_), axis=1)
+        # Step 2a: Estimate covariance of Y,Z
+        W = np.concatenate((self.residualized_Y_, self.residualized_Z_), axis=1)
         self.covfit_zy_ = self.covariance_estimator.fit(W)
+        cols_Y = self.residualized_Y_.shape[1]
+        self.cov_yz_ = self.covfit_zy_.covariance_[:cols_Y, cols_Y:]
 
-        cols_Z = self.residualized_Z_.shape[1]
-
-        self.cov_zz_ = self.covfit_zy_.covariance_[0 : cols_Z - 1, 0 : cols_Z - 1]
-        self.cov_yz_ = self.covfit_zy_.covariance_[cols_Z:, 0 : cols_Z - 1]
-        self.cov_yy_ = self.covfit_zy_.covariance_[cols_Z:, cols_Z:]
+        # Step 2b: Estimate precision of Z
+        self.covfit_zz_ = self.precision_estimator.fit(self.residualized_Z_)
+        self.prec_zz_ = self.covfit_zz_.precision_
 
-        # TODO : Use the precision estimator(s?) below:
-        # Estimate inverse of ZZ if dimensionality is small:
-        self.prec_zz_ = linalg.pinvh(self.cov_zz_, check_finite=False)
-        self.prec_yy_ = linalg.pinvh(self.cov_yy_, check_finite=False)
+        # Step 2c: Estimate precision of Y
+        self.covfit_yy_ = self.precision_estimator.fit(self.residualized_Y_)
+        self.prec_yy_ = self.covfit_yy_.precision_
 
+        # Step 2d: Calculate residual cross-covariance
         self.residual_crosscovariance_ = (
             (self.cov_yz_ @ self.prec_zz_) @ self.cov_yz_.T
         ) @ self.prec_yy_
```

Comment:
Shouldn't this now be:

```suggestion
        self.residual_crosscovariance_ = (
            (self.cov_yz_.T @ self.prec_zz_) @ self.cov_yz_
        ) @ self.prec_yy_
```

to match this eq? 
<img width="160" alt="Screen Shot 2021-07-02 at 10 20 37 AM" src="https://user-images.githubusercontent.com/118582/124309157-27046280-db1f-11eb-86a1-86741d1cc14d.png">

Reply:
I'm confused by this. If <img src="https://render.githubusercontent.com/render/math?math=T"> has dimensions <img src="https://render.githubusercontent.com/render/math?math=(n, q)">, and <img src="https://render.githubusercontent.com/render/math?math=\mathbf{S}"> has dimensions <img src="https://render.githubusercontent.com/render/math?math=(n, p)">, then isn't it the case that <img src="https://render.githubusercontent.com/render/math?math=\mathbf{\Sigma}_{\mathbf{S}T}"> has dimensions <img src="https://render.githubusercontent.com/render/math?math=(q, p)">, <img src="https://render.githubusercontent.com/render/math?math=\mathbf{\Sigma}_{\mathbf{SS}}^{-1}"> has dimensions <img src="https://render.githubusercontent.com/render/math?math=(p, p)">. If so then

<img src="https://render.githubusercontent.com/render/math?math=\gamma_{\Delta}^2 = \frac{\mathbf{\Sigma}_{\mathbf{S}T} \mathbf{\Sigma}_{\mathbf{SS}}^{-1} \mathbf{\Sigma}_{\mathbf{S}T}^{\dagger}}{\sigma_T}">

is the one that makes sense to me from a dimensional analysis perspective.